src/main/java/me/kavin/piped/utils/ResponseHelper.java

```diff
@@ -625,6 +625,52 @@ public static byte[] commentsPageResponse(String videoId, String prevpageStr) th
 
     private static final Argon2PasswordEncoder argon2PasswordEncoder = new Argon2PasswordEncoder();
 
+    public static byte[] deleteUserResponse(String session, String pass) throws IOException {
+
+        if (StringUtils.isBlank(pass))
+            return Constants.mapper.writeValueAsBytes(new InvalidRequestResponse());
+
+        try (Session s = DatabaseSessionFactory.createSession()) {
+            User user = DatabaseHelper.getUserFromSession(session);
+
+            if (user == null)
+                return Constants.mapper.writeValueAsBytes(new AuthenticationFailureResponse());
+            
+            String hash = user.getPassword();
+            boolean passMatch =
+                (hash.startsWith("$argon2") && argon2PasswordEncoder.matches(pass, hash)) || bcryptPasswordEncoder.matches(pass, hash);
+
+            if (!passMatch)
+                return Constants.mapper.writeValueAsBytes(new IncorrectCredentialsResponse());
+            
+            try {
+                CriteriaBuilder plCriteria = s.getCriteriaBuilder();
+                CriteriaQuery<me.kavin.piped.utils.obj.db.Playlist> plQuery =
+                    plCriteria.createQuery(me.kavin.piped.utils.obj.db.Playlist.class);
+                Root<me.kavin.piped.utils.obj.db.Playlist> plRoot =
+                    plQuery.from(me.kavin.piped.utils.obj.db.Playlist.class);
+                plQuery.select(plRoot).where(plCriteria.equal(plRoot.get("owner"), user.getId()));
+                List<me.kavin.piped.utils.obj.db.Playlist> playlists = s.createQuery(plQuery).getResultList();
+                
+                Iterator<me.kavin.piped.utils.obj.db.Playlist> iter = playlists.iterator();
+
+                while (iter.hasNext())
+                    s.delete(iter.next());
+
+                s.delete(user);

```

Comment:
Is the above iteration of playlists required, when we have this? I believe it should delete related objects but I may be wrong, I'll have a look at these later today!

Reply:
Yes, I just tested it. After commenting out the playlist deletion test and the remove from playlist test, so only the user is deleted, and removed the above iteration, the playlist is still in the database after deleting the user.
I think it's because the `Playlist` entity is not referenced by the `User` entity (i.e. from `User.java`).

Reply:
> I think it's because the Playlist entity is not referenced by the User entity (i.e. from User.java).

Maybe this is something that can be done instead :)

I'll try this out

Reply:
I've just added this, but I haven't tested it whether its actually deleted in the database.